src/event/sys/unix/parse.rs

```diff
@@ -312,7 +403,9 @@ pub(crate) fn parse_csi_u_encoded_key_code(buffer: &[u8]) -> Result<Option<Inter
         };
 
     let keycode = {
-        if let Some(c) = char::from_u32(codepoint) {
+        if let Some(special_key_code) = translate_functional_key_code(codepoint) {
```

Comment:
Before this statement, we gather modifier information. Can you perhaps check in this function if shift, alt, META were pressed and add this to the `KeyModifiers` bitflag?

Reply:
Just such that we dont get two seperate ways to tell the user about what modifier key was pressed